src/systems/audio-system.js

```diff
@@ -164,4 +152,49 @@ export class AudioSystem {
       this.audioNodes.delete(id);
     }
   }
+
+  /**
+   * Chrome and Safari will start Audio contexts in a "suspended" state.
+   * A user interaction (touch/mouse event) is needed in order to resume the AudioContext.
+   */
+  _resumeAudioContext = () => {
+    this.audioContext.resume();
+
+    setTimeout(() => {
+      if (this.audioContext.state === "running") {
+        if (!AFRAME.utils.device.isMobile() && /chrome/i.test(navigator.userAgent)) {
+          enableChromeAEC(this._sceneEl.audioListener.gain);
+        }
+
+        document.body.removeEventListener("touchend", this._resumeAudioContext, false);
+        document.body.removeEventListener("mouseup", this._resumeAudioContext, false);
+      }
+    }, 0);
+  };
+
+  // Webkit mobile fix
+  // https://stackoverflow.com/questions/10232908/is-there-a-way-to-detect-a-mobile-safari-audio-interruption-headphones-unplugg
+  _safariMobileAudioInterruptionFix() {
+    this.audioContext.onstatechange = () => {
+      console.log(`AudioContext state changed to ${this.audioContext.state}`);
+      if (this.audioContext.state === "suspended") {
+        // When you unplug the headphone or when the bluetooth headset disconnects on
+        // iOS Safari or Chrome, the state changes to suspended.
+        // Chrome Android doesn't go in suspended state for this case.
+        document.getElementById("avatar-rig").messageDispatch.receive({
+          type: "log",
+          messageType: LogMessageType.audioSuspended
+        });
```

Comment:
nit: Looks like MessageDispatch has a `log` method that you could use to abstract some of this away.

Reply:
True, fixed. Thanks!